driver-core/src/main/com/mongodb/internal/connection/BaseCluster.java

```diff
@@ -86,6 +87,8 @@
         this.serverFactory = notNull("serverFactory", serverFactory);
         this.clusterListener = getClusterListener(settings);
         clusterListener.clusterOpening(new ClusterOpeningEvent(clusterId));
+        description = new ClusterDescription(settings.getMode(), ClusterType.UNKNOWN, Collections.<ServerDescription>emptyList(),
```

Comment:
This is the root of it: The methods of this class assume that by the time the derived class constructors complete, `description` is non-null.  This chance guarantees it.

Reply:
If a derived construct DID assign to description doesn't this change overwrite it?

Reply:
Derived constructors run after base class constructors.  So a derived constructor could overwrite this (even set it to null), but that's its problem then.

Reply:
Shoot... I knew that! :) Sorry...